src/iuse.cpp

```diff
@@ -6459,20 +6559,800 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+                         " but <color_gray>inside</color> can be seen." );
+    }
+    photo_text += "\n" + overmap_desc + ".";
+
+    if( g->get_levz() >= 0 && need_store_weather ) {
+        photo_text += "\n\n";
+        if( calendar::turn.is_sunrise_now() ) {
+            photo_text += _( "It is <color_yellow>sunrise</color>. " );
+        } else if( calendar::turn.is_sunset_now() ) {
+            photo_text += _( "It is <color_light_red>sunset</color>. " );
+        } else if( calendar::turn.is_night() ) {
+            photo_text += _( "It is <color_dark_gray>night</color>. " );
+        } else {
+            photo_text += _( "It is day. " );
+        }
+
+        const weather_datum w_data = weather_data( g->weather.weather );
+        photo_text += string_format( _( "The weather is %s." ), colorize( w_data.name, w_data.color ) );
+    }
+
+    for( const auto &figure : description_figures_appearance ) {
+        photo_text += "\n\n<color_light_blue>" + figure.first + "</color>" + _( " appearance:" ) + "\n" +
+                      figure.second;
+    }
+
+    photo_text += "\n\n" + string_format( _( "The photo was taken on %s." ),
+                                          "<color_light_blue>" + timestamp + "</color>" );
+
+    photo.description = photo_text;
+
+    return photo;
+}
+
+static void item_save_monsters( player *p, item *it, const std::vector<monster *> &monster_vec,
+                                const int photo_quality )
+{
+    std::string monster_photos = it->get_var( "CAMERA_MONSTER_PHOTOS" );
+    if( monster_photos.empty() ) {
+        monster_photos = ",";
+    }
+
+    for( monster * const &monster_p : monster_vec ) {
+        const std::string mtype = monster_p->type->id.str();
+        const std::string name = monster_p->name();
+
+        // position of <monster type string>
+        const size_t mon_str_pos = monster_photos.find( "," + mtype + "," );
+
+        if( mon_str_pos == std::string::npos ) { // new monster
+            monster_photos += string_format( "%s,%d,", mtype, photo_quality );
+        } else { // replace quality character, if new photo is better
+            const size_t quality_num_pos = mon_str_pos + mtype.size() + 2;
+            char *quality_char = &monster_photos[ quality_num_pos ];
+            const int old_quality = atoi( quality_char ); // get qual number from char
+
+            if( photo_quality > old_quality ) {
+                monster_photos[ quality_num_pos ] = string_format( "%d", photo_quality )[ 0 ];
+            }
+            if( p && !p->is_blind() ) {
+                if( photo_quality > old_quality ) {
+                    p->add_msg_if_player( m_good, string_format(
+                                              _( "The quality of <color_light_blue>%s</color> image is better than the previous one." ), name ) );
+                } else if( old_quality == 5 ) {
+                    p->add_msg_if_player( string_format(
```

Comment:
`add_msg_if_player` does its own string formatting, no need to invoke that here manually:
```suggestion
                    p->add_msg_if_player(
```